stac_geoparquet/arrow/_batch.py

```diff
@@ -83,6 +87,9 @@ def from_dicts(
         # `ArrowInvalid: cannot mix list and non-list, non-null values`
         wkb_items = []
         for item in items:
+            if isinstance(item, pystac.Item):
```

Comment:
At some point we'll want to profile this. I'm guessing that the item copying and shapely stuff is going to be slower than `isinstance`, but it's a little unfortunate to pay the cost. We could avoid that with a `from_pystac_items` which takes a an `Iterable[pystac.Item]`.

Reply:
I can remove the pystac functionality from this PR if you prefer

Reply:
Took a quick look, and I think this is cheap enough relative to the other stuff to be fine:

```python
In [8]: import shapely, pystac

In [9]: item = pystac.read_file("https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20230112T104411_R008_T29NPE_20230113T053333")

In [11]: %timeit isinstance(item, pystac.Item)
64.2 ns ± 8.36 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [12]: %timeit shapely.to_wkb(shapely.geometry.shape(item.geometry), flavor="iso")
36.7 µs ± 4.78 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```